### ISSUE TITLE
We need instructions for installing mods

### ISSUE THREAD

**Issue by NeoRider7** [Tahvohck/Planetbase-Mods#1]
Hi!
I am grateful for the revival of Planetbase and that you made new mods!
But, unfortunately, I am not a programmer and it is not clear to me how to install your new mods.
Could you please provide instructions on how to install your new mods?
We need such instructions for installing mods that a person who is not a programmer can understand.
Thank you very much!

**Comment by Emperorderpintine** [comment]
Yes, please add some instructions on how to use this, not everyone knows how to do this, and even less know how to google to see what might need to be done. Can you also add some troubleshooting? I tried using Poweshell to get it to run, but it gave me lots of errors, the only things it said happened without errors were Generating Planetbase node... , Backed up config file, Saved config, and Script done

**Comment by NeoRider7** [comment]
For people far from programming, the instructions that Tahvohck wrote are not enough.
Ideally, it would be great to make a video with an example of how the engine from Tahvohck is installed and how the mod is then installed on this engine.

**Comment by NeoRider7** [comment]
The fact is that there are a huge number of messages from players that they do not understand at all how to install the engine and mods from Tahvohck.
For example, see what people write in the comments here:
Example-1: https://forums.nexusmods.com/index.php?showtopic=8777488/#entry96049513
Example-2: https://forums.nexusmods.com/index.php?showtopic=8777488/#entry89437523
Example-3: https://forums.nexusmods.com/index.php?showtopic=8777488/#entry87279838
Example-4: https://forums.nexusmods.com/index.php?showtopic=8777488/#entry82867548
Example-5: https://forums.nexusmods.com/index.php?showtopic=8777488/#entry82479628
Example-6: https://forums.nexusmods.com/index.php?showtopic=8827683/#entry87327203
...
It will be very cool if you reply to people on their messages :-)
Ideally, all gamers would like a video to be created showing how the engine is installed and then how mods are installed.

If you do not make the installation instructions understandable to ordinary players, then no one will ever be able to play your mods.  :-(

**Comment by Tahvohck** [comment]
unfortunately I remember the UMM documentation being pretty rough... And I'm not able to create videos either. I'll see if I can get it working again on my end and give a step-by-step on how to get it running, then someone else can make a video. That might take a while since I unfortunately don't have the steps I took in the past and UMM might have changed in the meantime, which would explain the errors that @Emperorderpintine were getting. It shouldn't have thrown errors at all, though it is pretty noisy about what it's doing.

**Comment by Tahvohck** [comment]
Well, this ended up being a bit more simple than I expected. Here's what the process should look like.

- Install UMM from their repository - you have to use the dropbox link for this, since they don't have releases: https://github.com/newman55/unity-mod-manager
- run `.\UnityModManagerSetup\install-planetbase-to-umm.ps1`. I thought you could do this via double-clicking, but it looks like you have to do it from a powershell window.
- In the file dialog that pops up, find UMM's directory and select the only file that's available. When you're done, you should see this.
![image](https://user-images.githubusercontent.com/1668722/151678447-bb771c4a-3b03-4c00-9045-b546d1a019cf.png)
- Run UMM and select Planetbase from the dropdown. It will be missing the game directory.
![image](https://user-images.githubusercontent.com/1668722/151678534-474441bc-b8cf-4dfd-8bb2-e2c469193dcf.png)
- Locate your planetbase game directory and install the DoorstopProxy
![image](https://user-images.githubusercontent.com/1668722/151678593-6046442f-3638-48ad-acbd-684ccc80b9bf.png)
- Switch to the Mods tab and drag-and-drop the mod zip files into that panel
![image](https://user-images.githubusercontent.com/1668722/151678634-228d045b-5bca-4243-90a3-bac89d161a41.png)
- Launch planetbase. It should automatically pop open the UMM menu in-game, showing the status of every loaded mod.
![image](https://user-images.githubusercontent.com/1668722/151678724-1b3058c6-32c4-4c00-9413-7413f0b1d072.png)

**Comment by Tahvohck** [comment]
Some other useful links for mod developers:
https://wiki.nexusmods.com/index.php/Category:Unity_Mod_Manager
https://github.com/newman55/unity-mod-manager/wiki/How-to-create-a-mod-for-unity-game

**Comment by NeoRider7** [comment]
> Well, this ended up being a bit more simple than I expected. Here's what the process should look like.
> 
> * Install UMM from their repository - you have to use the dropbox link for this, since they don't have releases: https://github.com/newman55/unity-mod-manager
> * run `.\UnityModManagerSetup\install-planetbase-to-umm.ps1`. I thought you could do this via double-clicking, but it looks like you have to do it from a powershell window.
> * In the file dialog that pops up, find UMM's directory and select the only file that's available. When you're done, you should see this.
> * Run UMM and select Planetbase from the dropdown. It will be missing the game directory.
> * Locate your planetbase game directory and install the DoorstopProxy
> * Switch to the Mods tab and drag-and-drop the mod zip files into that panel
> * Launch planetbase. It should automatically pop open the UMM menu in-game, showing the status of every loaded mod.

Thank you so much for posting information!
I have shown your instructions for installing your mod engine to several players.
But the players still said that you explain too hard. They haven't been able to install your mod engine yet.
Your explanation will probably be clear to cool programmers like yourself.
Please try to explain in such a way that a child understands, or at least an ordinary player in the game who has not encountered programming.

At the moment, even the first two steps in your instructions cause huge problems for ordinary people.
For instance,
**1) in your first paragraph of the instruction it says:**
"Install UMM from their repository - you have to use the dropbox link for this, since they don't have releases".
Firstly, if a person is not a programmer, then when he enters the github site, he is simply shocked and does not understand anything.
Secondly, not everyone understands what Dropbox is, how to find and install it.
In fact, you don’t need to install dropbox, but it seems that only you and I understand this. For ordinary people, this paragraph of your instructions is already baffling.
Especially when you consider that many players understand English very poorly.

What can be done better when writing the first paragraph of your instruction:
An example of how I would write the first paragraph of your instruction in other words:
"You need to download the latest version of the UMM program from the link: https://github.com/newman55/unity-mod-manager
This can be done, for example, in this way:
a) When you open this web page, you will see a green button at the top right with the inscription "Code".
b) By clicking on this green button you will see a menu item called "Download ZIP".
c) Download this zipped ZIP file.
d) Place this unpacked file at....
"
By the way, where should I place the unpacked file?

**2) in your second paragraph of the instruction it says:**
"run .\UnityModManagerSetup\install-planetbase-to-umm.ps1. I thought you could do this via double-clicking, but it looks like you have to do it from a powershell window."
What is PowerShell? Where and how to download PowerShell?

In general, out of many players to whom I showed your instructions, so far not an oldin could figure it out and install any mod on your mod engine.

Please make the instructions easy to understand for regular gamers, not just cool programmers like you.

And it is **highly desirable that you place this instruction on your websites on the NexusMods resource:**
1) "Unity Mod Manager Configuration" https://www.nexusmods.com/planetbase/mods/43
2) "Planetbase Utilities" https://www.nexusmods.com/planetbase/mods/45
3) "AutoConnections ReManaged" https://www.nexusmods.com/planetbase/mods/46
4) Building Aligner ReManaged https://www.nexusmods.com/planetbase/mods/47

By the way, please go to your page on the NexusMods resource and look at the "[POSTS](https://www.nexusmods.com/planetbase/mods/43?tab=posts)" sections.
There players have written **a lot of questions for you** over the last couple of years.

We all really hope that we will understand how to install your mods and be able to finally play them!
Thank you very much for everything!

**Comment by Tahvohck** [comment]
Please keep in mind that my code isn't actually a mod engine. That heavy lifting is done by the Unity Mod Manager itself, which is a generalized mod injector for any unity game. All of my code is only mods, even PBUtilities (which is a mod some of the other mods depend on). As a more technically inclined person who's still interested in the PB modding scene, I'm hoping to leave the generalized user-friendly language to you. Unfortunately I got pretty busy and lost interest in planetbase more than a year ago (which is why this repository is in such a sorry state), so my ability to provide direct support is pretty limited.

With that said, I do see some things need to be clarified. 

1) You shouldn't be using the code link in UMM's repo: that's  but the dropbox link in their readme. I don't like how they've done this, it's not standard, but there's not much we can do about it. The problem is that they might change the readme at some point in the future, making specific instructions potentially fragile. For now, the specific [link is here](https://www.dropbox.com/s/wz8x8e4onjdfdbm/UnityModManager.zip?dl=1). That's the fully compiled program for UMM, you can put it wherever you want.
2) Powershell is the modern Windows command line. You don't need to download it, it's built into Windows. Unfortunately I'm not familiar at all with UI programming, so unless someone else makes it, the fairly user-unfriendly script is what the community is stuck with. Still, it's better than telling people to manually edit the config file for UMM. There's also the option of submitting a code request to UMM directly, something I really should have way back when I worked on this. Which... I'll try to do after submitting this comment, but with how long it's been since THEY updated their repo, I don't have high hopes.

Once some user-friendly instructions or a video is made, I'd be happy to add it to all the relevant mods.

I'm really glad to see the modding scene for planetbase being revived, but unfortunately I can't be part of it going forward beyond helping you guys get it running here.
